axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineToolServiceImpl.java

```diff
@@ -358,4 +365,48 @@ public void checkDateInPeriod(Move move, MoveLine moveLine) throws AxelorExcepti
       }
     }
   }
+
+  @Transactional
+  public void setAmountRemainingReconciliableMoveLines(String startDate, Long accountId) {
+    List<Map> idMoveLinesList;
+    int count = 0;
+
+    if (startDate == null) {
+      idMoveLinesList =
+          this.moveLineRepository
+              .all()
+              .filter("self.accountId = :accountId")
+              .bind("accountId", accountId)
+              .order("id")
+              .select("id")
+              .fetch(0, 0);
+    } else {
+      idMoveLinesList =
+          this.moveLineRepository
+              .all()
+              .filter("self.accountId = :accountId AND self.move.createdOn >= :startDate")
+              .bind("accountId", accountId)
+              .bind("startDate", startDate)
+              .order("id")
+              .select("id")
+              .fetch(0, 0);
+    }
+
+    if (idMoveLinesList != null) {
+      while (count < idMoveLinesList.size()) {
+        MoveLine moveLine =
+            this.moveLineRepository.find(
+                parseLong(idMoveLinesList.get(count).get("id").toString()));
+        moveLine.setAmountRemaining(
+            BigDecimal.valueOf(
+                max(moveLine.getDebit().longValue(), moveLine.getCredit().longValue())));
+        this.moveLineRepository.save(moveLine);
+        count++;
+
+        if (count % jpaLimit == 0) {
```

Comment:
Nice, you thought about the JPA issue, the problem is, you put a @Transactional on your method, so the JPA.clear will erase everything and nothing will be commited at the end, because with a @Transactional, nothing is commited/persisted in the database before you leave the method.

You won't need to do a JPA.clear anymore, because with the update, everything is done directly in the database.

Reply:
Using fetch, we need to solve JPA issue.
Refactor of this part